src/ngraph/pass/nop_elimination.cpp

```diff
@@ -134,62 +134,235 @@ static bool eliminate_reshape_v1(const std::shared_ptr<Node>& node)
     return false;
 }
 
+static bool replace_squeeze_unsqueeze(const std::shared_ptr<Node>& node)
+{
+    auto out_shape = node->get_output_partial_shape(0);
+    if (!out_shape.is_dynamic() && shape_size(out_shape.get_shape()) != 0)
+    {
+        shared_ptr<Node> reshape;
+        auto input = node->input_value(0).get_node_shared_ptr();
+        auto shape = node->get_shape();
+        std::vector<int64_t> vi;
+        vi.assign(shape.begin(), shape.end());
+        auto pat = op::Constant::create<int64_t>(element::i64, Shape{vi.size()}, vi);
+
+        if (as_type_ptr<opset3::Reshape>(input) || as_type_ptr<opset3::Squeeze>(input) ||
+            as_type_ptr<opset3::Unsqueeze>(input))
+        {
+            input = input->input_value(0).get_node_shared_ptr();
+            reshape = make_shared<opset3::Reshape>(input->output(0), pat, false);
+        }
+        else
+        {
+            reshape = make_shared<opset3::Reshape>(input->output(0), pat, false);
+        }
+
+        // skip if reshape is nop
+        if (reshape->get_input_shape(0) == shape)
+        {
+            return replace_output_update_name(node->output(0), reshape->input_value(0));
+        }
+        else
+        {
+            return replace_output_update_name(node->output(0), reshape->output(0));
+        }
+    }
+    return false;
+}
+
+static std::vector<int64_t> get_unsqueeze_axes(const PartialShape& data_shape,
+                                               const PartialShape& out_shape)
+{
+    std::vector<int64_t> axes;
+    size_t i = 0;
+    for (auto o = 0; o < out_shape.rank().get_length(); o++)
+    {
+        if (i < data_shape.rank().get_length() && data_shape[i].same_scheme(out_shape[o]))
+        {
+            i += 1;
+            continue;
+        }
+        if (out_shape[o].is_static() && out_shape[o] == 1)
```

Comment:
What if out_shape[o] is non-static, the axes won't be correct. Does this case exist?

Reply:
we treat dynamic (non-static) dimension as if they are not "1". So squeeze/unsqueeze will not be done for them.